### ISSUE TITLE
[Idea] Defining Custom Mobs

### ISSUE THREAD

**Issue by Thaumcraft4Lyfe** [garbagemule/MobArena#351]
I think a new feature should be added: Custom Mobs. These could be equipped with armor, wear a block or player head as a hat, be equipped with different weapons, and support custom health, damage, and abilities. 

This could potentially work by adding a new config file, a custom-mobs.yml. This would allow you to define a new mob, in a format similar to this: (Just an example)

<mobname>:
    mob: <the type of mob it is>
    name: <display name>
    health: <hp>
    damage: <default dmg on hit>
    armor: <any armor it is wearing, uses a format similar to class armor>
    hat: <if there is no helmet defined, you can equip a block or playerhead by stating the name of the player or the block/item id>
    holding: <a weapon or item it is holding>
    holding-enchants: <any enchantments you want it to be holding, formatted like name:level>
    abilities: <This would allow you to select boss abilities for this mob to cast. Certain abilities, like the ones that teleport players would not work>
    cooldown: <the time in between abilities>    

When selecting a mob type in the wave setup, you would use "custom:<mobname>" instead.

This would allow for a new range of possibilities. I think it would fit well in this plugin. Thanks.

**Comment by garbagemule** [comment]
We need lots of ideas and feedback on this before we can even attempt a stab at this.

Also ties in really well with the rework of monster spawning/managing.

**Comment by mibby** [comment]
Perhaps add support for hooking into MythicMobs & Boss for custom mob spawns? I personally use the latter plugin.

https://www.spigotmc.org/resources/%E2%9A%94-mythicmobs-free-version-%E2%96%BAthe-1-custom-mob-creator%E2%97%84.5702/
https://www.spigotmc.org/resources/%E2%98%9B-boss-unbelievable-custom-monsters.46497/

**Comment by garbagemule** [comment]
MythicMobs support is already tracked on #362

While I like the idea of combining plugins for a more fine-tuned experience, direct plugin integration is high maintenance and usually reduces to some sort of lowest common denominator, where a really great feature of one of the plugins isn't available when combining with the other plugin. I think MythicMobs and Boss are cool plugins in their own right, but they are not designed to be used in a minigame setting. Of the two, MythicMobs is by far the most elaborate, and integration does have some potential that might be worth exploring in the sessions/mobs rework. Boss, on the other hand, is technically just MobArena bosses with a small amount of customization.

MobArena is about mobs and arenas. Those two should be what it does well and on its own. A rework of these core concepts is long overdue, but we have also learned _a lot_ over the years, both about the Minecraft/MobArena landscape, and as contributors to the project. My envisioned rework of mobs and sessions should expand the experience in the direction of complex "encounters", rather than having a hard focus on individual mobs and set-in-stone wave progressions.
